### PR TITLE
Adjustments to actions operator on ubuntu-24.04

### PR DESCRIPTION
**Pull request by addyess**
### Overview

* Addresses issue #82 by testing CI with 24.04 
* Apply `docker_lxd_clash` workaround for jammy (22.04.1) and beyond
* working around issue with sg with sudo
    * https://github.com/actions/runner-images/issues/9932
* addressing issues installing PyYAML==6.0

### CODE REVIEW

File: src/bootstrap/index.ts [hunk 1]
```diff
@@ -120,15 +141,11 @@ async function microk8s_init(channel, addons, container_registry_url:string) {
     // Add the given addons if any were given.
     await exec_as_microk8s("microk8s status --wait-ready --timeout=600");
     if (addons) {
-        await exec_as_microk8s("sudo microk8s enable " + addons);
+        await exec.exec(`sudo microk8s enable ${addons}`);
```
Comment (mateoflorido) [inline src/bootstrap/index.ts:144]: Shouldn't this be `exec_as_microk8s` as well?

**Review by mateoflorido — APPROVED** [review]
LGTM! Just a couple of nits

File: src/bootstrap/index.ts (hunk 1 above)
Comment (addyess) [reply src/bootstrap/index.ts:144]: so i tried 

```typescript
await exec_as_microk8s("microk8s enable " + addons);
```

but I was prompted with a "you need sudo for that".  I think it means that the bash script for one of the addons actually needed `sudo`.  I decided to just drop trying to sudo wrap another sudo call...

File: src/bootstrap/index.ts [hunk 2]
```diff
@@ -29,6 +31,25 @@ const os_release = async () => {
...
 }
 
+const snap_version = async (snap_name: string) => {
+    let stdout_buf = '';
+    const options = {
+        listeners: {
+            stdout: (data) => { stdout_buf += data.toString() }
+        }
+    };
+    await exec.exec("snap", ["list", snap_name], options);
```
[the remark this answers is not in the document]
Comment (addyess) [reply src/bootstrap/index.ts:41]: i'm not super sure of it either.  I'd almost rather the stack traces flow.  I'd rather not try to handle anything that we might catch... is there something you think might be worth catching here?

[the remark this answers is not in the document]
Comment (mateoflorido) [reply src/bootstrap/index.ts:41]: That sounds good, let's keep the stack traces.

File: src/bootstrap/index.ts (hunk 1 above)
Comment (mateoflorido) [reply src/bootstrap/index.ts:144]: Cool, thanks for the explanation :)

**Comment by addyess** [comment]
@jnsgruk I believe these changes resolve your issue.  Would you like to test from this branch or are you +1 on these changes without testing?

**Comment by jnsgruk** [comment]
Opened https://github.com/canonical/parca-k8s-operator/pull/346 -- if that looks good then let's get this merged. Thank you!

**Comment by jnsgruk** [comment]
@addyess - not quite, it seems: https://github.com/canonical/parca-k8s-operator/actions/runs/11021119226/job/30607505769?pr=346

**Review by jnsgruk — APPROVED** [review]

**Comment by jnsgruk** [comment]
Okay, we're good I think, thanks for this!

**Pull request merged**